kitsune/search/v2/search.py

```diff
@@ -0,0 +1,238 @@
+        pass
+
+
+class AAQConfig(BaseConfig):
+    """Search config for AAQ."""
+
+    index = QUESTION_INDEX_NAME
+    fields = [
+        "question_title.{}^4",
+        "question_content.{}^3",
+        "answer_content.{}^3",
+    ]
+    highlight_fields = ["question_content.{}", "answer_content.{}"]
+
+    @property
+    def filter(self):
+        return Q(
+            "bool",
+            filter=super().filter,
+            # exclude AnswerDocuments from the search:
+            must_not=Q("exists", field="updated"),
+        )
+
+    def product_filter(self, product):
+        return Q("term", question_product_id=product.id)
+
+    def make_result(self, hit, locale):
+        summary = first_highlight(hit)
+        if not summary:
+            summary = hit.question_content[locale][:500]
+        summary = sanitize(summary)
+
+        num_answers = 0
+        answer_content = getattr(hit, "answer_content", None)
+        if answer_content:
+            num_answers = len(answer_content[locale])
+
+        return {
+            "type": "question",
+            "url": "/questions/{}".format(hit.question_id),
+            "score": hit.meta.score,
+            "title": hit.question_title[locale],
+            "search_summary": summary,
+            "last_updated": datetime.fromisoformat(hit.question_updated),
+            "is_solved": hit.question_has_solution,
+            "num_answers": num_answers,
+            "num_votes": hit.question_num_votes,
+        }
+
+
+class KBConfig(BaseConfig):
+    """Search config for the Knowledge Base."""
+
+    index = WIKI_DOCUMENT_INDEX_NAME
+    fields = [
+        "keywords.{}^8",
+        "title.{}^6",
+        "summary.{}^2",
+        "content.{}^1",
+    ]
+    highlight_fields = ["summary.{}", "content.{}"]
+
+    @property
+    def filter(self):
+        return Q(
+            "bool",
+            filter=super().filter,
+            must_not=Q("terms", category=[TEMPLATES_CATEGORY, CANNED_RESPONSES_CATEGORY]),
+        )
+
+    def product_filter(self, product):
+        return Q("term", product_ids=product.id)
+
+    def make_result(self, hit, locale):
+        summary = first_highlight(hit)
+        if not summary and hasattr(hit, "summary"):
+            summary = getattr(hit.summary, locale, None)
+        if not summary:
+            summary = hit.content[locale][:500]
+        summary = sanitize(summary)
+
+        return {
+            "type": "document",
+            "url": "/{}/kb/{}".format(locale, hit.slug[locale]),
+            "score": hit.meta.score,
+            "title": hit.title[locale],
+            "search_summary": summary,
+        }
+
+
+class Search:
+    """Create an ES7 Search."""
+
+    def __init__(self, query, language, product=None, page=1):
+        self.config_classes = []
+        self.query = query
+        self.language = language
+        self.product = product
+        self.page = page
+
+    def add_config(self, config):
```

Comment:
`config` feels wrong here. Maybe search_type ? Feel free to add something better since the proposed one is not great either